cmd/dep/gopath_scanner.go

```diff
@@ -53,8 +53,9 @@ func (g *gopathScanner) InitializeRootManifestAndLock(rootM *dep.Manifest, rootL
 	}
 
 	g.origM = &dep.Manifest{
-		Constraints: g.pd.constraints,
-		Ovr:         make(gps.ProjectConstraints),
+		Constraints:  g.pd.constraints,
+		Ovr:          make(gps.ProjectConstraints),
+		PruneOptions: gps.PruneNestedVendorDirs,
```

Comment:
Now that we want to have a default value for manifests (pruning nested vendor dirs), I'd prefer that we put that logic into a constructur `dep.NewManifest`, instead of relying upon every caller to remember to set it and get the default right.

Also, I didn't see that you updated the importers (they all make new manifests) and it's not clear to me what will even happen in that case if they leave the prune option off?